images/linux/scripts/installers/python.sh

```diff
@@ -1,27 +0,0 @@
-#!/bin/bash
-################################################################################
-##  File:  python.sh
-##  Desc:  Installs Python 2/3
-################################################################################
-
-# Source the helpers for use with the script
-source $HELPER_SCRIPTS/document.sh
-
-# Install Python, Python 3, pip, pip3
-apt-get install -y --no-install-recommends python python-dev python-pip python3 python3-dev python3-pip
```

Comment:
`python-dev` installs the .so and header files for developing C extensions for Python.  The tool cache versions have these installed as well.  I think `python setup.py` will handle setting the include file and library paths when building an extension, so hopefully this just works with the change.

I had [added a canary build](https://github.com/microsoft/azure-pipelines-canary/pull/114) for building C extensions with the versions in tool cache.  Could you make sure this works without running `Use Python Version` as well?